addons/html_editor/static/tests/html_field.test.js

```diff
@@ -1055,6 +1057,101 @@ test("'Video Link' command is available", async () => {
     expect(queryAllTexts(".o-we-command-name")).toEqual(["Video Link"]);
 });
 
+test.tags("desktop");
+test("should display overlay on video hover and handle video replacement and removal correctly", async () => {
+    await mountView({
+        type: "form",
+        resId: 1,
+        resModel: "partner",
+        arch: `
+            <form>
+                <field name="txt" widget="html"/>
+            </form>`,
+    });
+    setSelectionInHtmlField();
+
+    let callCount = 0;
+    await onRpc("/web_editor/video_url/data", async () => {
+        callCount++;
+        if (callCount === 1) {
+            return {
+                video_id: "qxb74CMR748",
+                platform: "youtube",
+                embed_url: "https://www.youtube.com/embed/qxb74CMR748?rel=0&autoplay=0",
+                params: {
+                    rel: 0,
+                    autoplay: 0,
+                },
+            };
+        } else {
+            return {
+                video_id: "gbE3azm_Io0",
+                platform: "youtube",
+                embed_url: "https://www.youtube.com/embed/gbE3azm_Io0?rel=0&autoplay=0",
+                params: {
+                    rel: 0,
+                    autoplay: 0,
+                },
+            };
+        }
+    });
+
+    // Insert video
+    await insertText(htmlEditor, "/video");
+    await waitFor(".o-we-powerbox");
+    expect(queryAllTexts(".o-we-command-name")[0]).toBe("Video Link");
+
+    await press("Enter");
+    await animationFrame();
+
+    const input = queryOne("textarea[id='o_video_text']");
+    input.value = "https://www.youtube.com/embed/qxb74CMR748?rel=0&autoplay=0";
+    manuallyDispatchProgrammaticEvent(input, "input", {
+        inputType: "insertText",
+    });
+    await animationFrame();
+    await waitFor(".o_video_dialog_options", { timeout: 1500 });
+    await click(".modal-footer button");
+    await animationFrame();
+    const iframeBefore = queryOne("div[data-embedded='video'] iframe");
+
+    // Hover on VideoBlock shows overlay
+    await hover(queryOne("div[data-embedded='video']"));
+    await animationFrame();
+    expect(queryOne(".video-overlay")).toHaveCount(1);
```

Comment:
Either use `expectElementCount` for this one, or do `waitFor` with `.video-overlay` instead of `animationFrame`, as I fear waiting one animation frame might not be enough when the runbot is overloaded and it might lead to non-deterministic test failures. See https://github.com/odoo/odoo/pull/211426/commits/54da715df84789f9a1acc0cfc91be41dcdbab140 for more details.